
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="ChIP-seq tutorial, University of Luxembourg, Department Life Sciences and Medecine">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.2, mkdocs-material-7.2.4">
    
    
      
        <title>Mapping - ChIP-seq tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.f7f47774.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#paleomix-next-generation-sequencing-wrapper" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq tutorials" class="md-header__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq tutorials
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Mapping
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq tutorials" class="md-nav__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    ChIP-seq tutorials
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../cli/" class="md-nav__link">
        Command line, basics
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../snakemake/" class="md-nav__link">
        Setup snakemake
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../workflow/" class="md-nav__link">
        Workflow
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Mapping
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Mapping
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#paleomix-next-generation-sequencing-wrapper" class="md-nav__link">
    paleomix, Next-Generation Sequencing wrapper
  </a>
  
    <nav class="md-nav" aria-label="paleomix, Next-Generation Sequencing wrapper">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#test-your-install" class="md-nav__link">
    test your install
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#generate-a-makefile" class="md-nav__link">
    Generate a makefile
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edit-the-makefile" class="md-nav__link">
    Edit the makefile
  </a>
  
    <nav class="md-nav" aria-label="Edit the makefile">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features" class="md-nav__link">
    Features
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prefixes" class="md-nav__link">
    Prefixes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samples" class="md-nav__link">
    Samples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-the-trimming-mapping" class="md-nav__link">
    Perform the trimming / mapping
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct-the-makefile" class="md-nav__link">
    correct the makefile
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#check-trimming" class="md-nav__link">
    check trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filter-for-unique-reads" class="md-nav__link">
    filter for unique reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filter-for-duplicates" class="md-nav__link">
    filter for duplicates?
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../peak/" class="md-nav__link">
        Peak calling
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../contact/" class="md-nav__link">
        Contact
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#paleomix-next-generation-sequencing-wrapper" class="md-nav__link">
    paleomix, Next-Generation Sequencing wrapper
  </a>
  
    <nav class="md-nav" aria-label="paleomix, Next-Generation Sequencing wrapper">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#test-your-install" class="md-nav__link">
    test your install
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#generate-a-makefile" class="md-nav__link">
    Generate a makefile
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edit-the-makefile" class="md-nav__link">
    Edit the makefile
  </a>
  
    <nav class="md-nav" aria-label="Edit the makefile">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#options" class="md-nav__link">
    Options
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features" class="md-nav__link">
    Features
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#prefixes" class="md-nav__link">
    Prefixes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#samples" class="md-nav__link">
    Samples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-the-trimming-mapping" class="md-nav__link">
    Perform the trimming / mapping
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#correct-the-makefile" class="md-nav__link">
    correct the makefile
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#check-trimming" class="md-nav__link">
    check trimming
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filter-for-unique-reads" class="md-nav__link">
    filter for unique reads
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filter-for-duplicates" class="md-nav__link">
    filter for duplicates?
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/ginolhac/chip-seq/edit/master/docs/mapping.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  <h1>Mapping</h1>
                
                <h2 id="paleomix-next-generation-sequencing-wrapper">paleomix, Next-Generation Sequencing wrapper</h2>
<p>this framework is open-source and available on <a href="https://github.com/MikkelSchubert/paleomix">GitHub</a> and 
wraps all steps from <code>fastq</code> to <code>bam</code> files. 
Actually, this tool can do much more but the rest is out of scope. 
Its major drawback is that it is dedicated to one machine. For clusters, you are then limited to one node since memory are not shared by default.
Actually, not entirely true since independent tasks can be spawn on a separate machine.
Full documentation available <a href="http://paleomix.readthedocs.io/en/latest/">here</a></p>
<p>check if paleomix is available</p>
<pre><code>paleomix
</code></pre>
<p>In case it is not, your are certainly not using the <em>singularity container</em>, see instructions in the <a href="http://ginolhac.github.io/chip-seq/install/">set-up</a></p>
<h3 id="test-your-install">test your install</h3>
<p>fetch the example, reference is the human mitochondrial genome</p>
<pre><code>
mkdir -p ~/paleomix/example
cp -r /scratch/users/aginolhac/chip-seq/bam_pipeline_example/000* ~/paleomix/example
cd ~/paleomix/example
</code></pre>
<p>run the example, start by a <code>dry-run</code>, adjust the number of threads accordingly.</p>
<pre><code>paleomix bam_pipeline run --bwa-max-threads=1 --max-threads=8 --dry-run 000_makefile.yaml
</code></pre>
<p>If all fine, re-rerun the command without the <code>--dry-run</code> option</p>
<p>Of note, calling <code>mapDamage</code> and <code>GATK</code> were disabled to limit the computation time (~ 35 min when included).
Anyway, this tool is not used for ChIP-seq analysis.</p>
<h3 id="generate-a-makefile">Generate a makefile</h3>
<p>Trimming, mapping imply a lot of steps and it is hard to be sure that everything goes well. 
Paleomix works in temporary folders, check the data produced and then copy back files that are complete. 
Plus, you want to test different parameters, add a new reference without having to redo earlier steps while being sure that all files are up-to-date. 
This goes through a <code>YAML</code> makefile. The syntax is pretty straight-forward.
What matters is, that you use <strong>SPACES</strong> and not TABS.</p>
<p>Create a generic makefile (extension, <code>yml</code> or <code>yaml</code> to get syntax highlights)</p>
<pre><code>cd ~/chip-seq
paleomix bam_pipeline mkfile &gt; mouse.yaml
</code></pre>
<h3 id="edit-the-makefile">Edit the makefile</h3>
<p>using your favorite text editor (such as <code>nano</code>), edit the <code>mouse.yaml</code>. For example <code>vim mouse.yaml</code> or <code>kate</code> or <code>nano</code>.</p>
<h4 id="options">Options</h4>
<ul>
<li>for the compression, change the default behavior from <code>bz2</code> to <code>gz</code>  </li>
</ul>
<pre><code>  CompressionFormat: gz
</code></pre>
<ul>
<li>for duplicates, change the default behavior from <code>filter</code> to <code>mark</code>  </li>
</ul>
<pre><code>  PCRDuplicates: mark
</code></pre>
<h4 id="features">Features</h4>
<p>Under the <code>Features</code> section, update the featurse that need to be performed.
Change <code>yes/no</code> to match the following:</p>
<pre><code>  Features:
    RawBAM: yes         # Generate BAM from the raw libraries (no indel realignment)
                        #   Location: {Destination}/{Target}.{Genome}.bam
    RealignedBAM: no    # Generate indel-realigned BAM using the GATK Indel realigner
                        #   Location: {Destination}/{Target}.{Genome}.realigned.bam
    mapDamage: no       # Generate mapDamage plot for each (unrealigned) library
                        #   Location: {Destination}/{Target}.{Genome}.mapDamage/{Library}/
    Coverage: yes       # Generate coverage information for the raw BAM (wo/ indel realignment)
                        #   Location: {Destination}/{Target}.{Genome}.coverage
    Depths: no          # Generate histogram of number of sites with a given read-depth
                        #   Location: {Destination}/{Target}.{Genome}.depths
    Summary: yes        # Generate summary table for each target
                        #   Location: {Destination}/{Target}.summary
    DuplicateHist: no   # Generate histogram of PCR duplicates, for use with PreSeq
                        #   Location: {Destination}/{Target}.{Genome}.duphist/{Library}/
</code></pre>
<p>In detail, the <code>RealignedBAM</code> are important for calling variants, we only need to <code>RawBAM</code>.
Moreover, the <code>Depths</code> also help to define which upper limit could be used for variant calling.
This is not in the scope of ChIP-seq analysis. Same for <code>mapDamage</code>, only relevant for ancient DNA.</p>
<h4 id="prefixes">Prefixes</h4>
<p>These are the references to align read to. You could notice that we are going to use only one chromosome
to save computational time.</p>
<pre><code>Prefixes:
  mouse_19:
    Path: /scratch/users/aginolhac/chip-seq/references/chr19.fasta
</code></pre>
<h4 id="samples">Samples</h4>
<p>enter at the end of the makefile, the following lines.
Again, do use <strong>spaces</strong> and not tabs for the indentation. For those who are lazy and use copy/paste in <code>vim</code>
 use the trick to <code>:set paste</code> to avoid extra spaces, comment hashes etc to be automatically added.</p>
<p>The descriptions of the different hierachical names
can be read <a href="http://paleomix.readthedocs.io/en/latest/bam_pipeline/makefile.html#targets-section">here</a></p>
<pre><code>TC1-I-A-D3:
  TC1-I-A-D3:
    TC1-I-A-D3:
      &quot;14s006680-1-1&quot;: fastq/C53CYACXX_TC1-I-A-D3_14s006682-1-1_Sinkkonen_lane114s006682_sequence.txt.gz

TC1-H3K4-A-D3:
  TC1-H3K4-A-D3:
    TC1-H3K4-A-D3:
      &quot;14s006647-1-1&quot;: fastq/C51C3ACXX_TC1-H3K4-A-D3_14s006647-1-1_Sinkkonen_lane514s006647_sequence.txt.gz

TC1-I-ST2-D0:
  TC1-I-ST2-D0:
    TC1-I-ST2-D0:
      &quot;14s006677-1-1&quot;: fastq/C51C3ACXX_TC1-I-ST2-D0_14s006677-1-1_Sinkkonen_lane814s006677_sequence.txt.gz

TC1-H3K4-ST2-D0:
  TC1-H3K4-ST2-D0:
    TC1-H3K4-ST2-D0:
      &quot;14s006644&quot;: fastq/C51C3ACXX_TC1-H3K4-ST2-D0_14s006644-1-1_Sinkkonen_lane514s006644_sequence.txt.gz
</code></pre>
<h3 id="perform-the-trimming-mapping">Perform the trimming / mapping</h3>
<p>First use the option <code>--dry-run</code> to spot mistakes.</p>
<p>Please <strong>adapt</strong> the <code>--max-threads</code> option to the #cpus actually booked</p>
<pre><code>paleomix bam_pipeline run --bwa-max-threads=2 --adapterremoval-max-threads=2 --max-threads=8 --dry-run mouse.yaml
</code></pre>
<p>when all green lights are on, remove the <code>dry-run</code> and perform the mapping.</p>
<h2 id="correct-the-makefile">correct the makefile</h2>
<p>the trimming should go well, but an error arises because you cannot write to the reference folder.</p>
<ul>
<li>symbolic links in your own folder, removing the file that should be created if the ref is correct</li>
</ul>
<pre><code>mkdir references
ln -s /scratch/users/aginolhac/chip-seq/references/chr19.fasta references/
</code></pre>
<ul>
<li>correct the makefile, so the <code>Path</code> is relative now</li>
</ul>
<pre><code>    Path: references/chr19.fasta
</code></pre>
<ul>
<li>enjoy <code>paleomix</code> by just re-running it, only necessary steps are done (validatating the ref, indexing it and mappings)</li>
</ul>
<pre><code>paleomix bam_pipeline run --bwa-max-threads=2 --adapterremoval-max-threads=2 --max-threads=8 mouse.yaml
</code></pre>
<p>the whole process should takes ~ 25 minutes with 8 cores</p>
<h2 id="check-trimming">check trimming</h2>
<p>First of all, check using <code>fastqc</code> that the trimming did remove the adapters that were contaminated the reads.</p>
<p>Again, with <code>parallel</code> specify the <strong>max</strong> number of jobs with the option <code>-j</code> to fit the #cpus booked</p>
<pre><code>find . -name &quot;reads.truncated.gz&quot; | parallel -j 8 &quot;fastqc {}&quot; &amp;
</code></pre>
<p>using the character <code>&amp;</code> tells the shell that we want the processes to run in the background. Meaning that you can still run more things while the 4 tasks are running. Check them using <code>htop</code>.</p>
<p>check especially, the input for ST2, day0 before and after trimming. Did it solve the issue with adapters?</p>
<p>truncated read files are all named the same. And they are located into a deeper folder structure. To rename them all with their ids and move them at the <code>~/chip-seq/</code> level, you can run the following:</p>
<pre><code>find -name &quot;reads.truncated.gz&quot; |  xargs ls -1 | awk '{split($1, path, &quot;/&quot;);  system(&quot;mv &quot;$0 &quot; &quot;path[3] &quot;_&quot; path[6])}'
</code></pre>
<h2 id="filter-for-unique-reads">filter for unique reads</h2>
<p><em>Uniqueness</em> of reads refers to mappability. The fewer locations a read has in a genome, the higher is mappability will be.
A common filter is to use <strong>30</strong> as a threshold for filtering reads. Filter them in parallel</p>
<pre><code>parallel &quot;samtools view -b -q 30 {} &gt; {.}.q30.bam&quot; ::: *.bam
</code></pre>
<p>Since we are using only the chr19 for this tutorial, do you think the mappability score is correct? Why?</p>
<h2 id="filter-for-duplicates">filter for duplicates?</h2>
<p>Duplication is a bias that comes from PCR amplification. Reads then stack at the same location and create artificial high depth of coverage.
Duplicates have an unclear definition in a mapped file. Usually, single-end reads that are mapped
at the same 5' end are considered as duplicates. External coordinates are used for paired-end reads.<br />
For regular NGS, filtering for duplicates is mandatory. However, for ChIP-seq since the reads are,
by nature, clustered at one location this is not recommended. If duplication is observed at the reads level, 
such as in <code>fastqc</code> output, then filtering may be necessary. Marking duplicates allow keeping track of them without losing them.</p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../workflow/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Workflow" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Workflow
            </div>
          </div>
        </a>
      
      
        
        <a href="../peak/" class="md-footer__link md-footer__link--next" aria-label="Next: Peak calling" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Peak calling
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.709b4209.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.febc23d1.min.js"></script>
      
    
  </body>
</html>