<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>peak(Conflicted) - ChIP-seq tutorials</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "peak(Conflicted)";
    var mkdocs_page_input_path = "peak(Conflicted).md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> ChIP-seq tutorials</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../install/">Setup</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../cli/">Command line, basics</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../fastqc/">QC</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../mapping/">Mapping</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../peak/">Peak calling</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../public/">Public data</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../contact/">Contact</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">ChIP-seq tutorials</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>peak(Conflicted)</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ginolhac/chip-seq/edit/master/docs/peak(Conflicted).md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h2 id="peak-calling">Peak calling</h2>
<p>Using <a href="https://github.com/taoliu/MACS/">MACS2</a></p>
<p>For both the day 0 and day 3 of differentiation into adipocytes, two files are available</p>
<ul>
<li>input, as control</li>
<li>histone modification H3K4</li>
</ul>
<p><strong>MACS2</strong> is going to use both files to normalize the read counts and perform the peak calling.</p>
<h3 id="retrieve-the-bam-files-with-all-chromosomes">Retrieve the BAM files with all chromosomes</h3>
<pre><code>cd ~/chip-seq
mkdir bams
cd bams
ln -s /scratch/users/aginolhac/chip-seq/bams/*.bam .
</code></pre>

<h3 id="perform-peak-calling">Perform peak calling</h3>
<pre><code>macs2 callpeak -t TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-ST2-H3K4-D0 -B -q 0.01 --outdir TC1-ST2-H3K4-D0 &amp;
macs2 callpeak -t TC1-H3K4-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-A-H3K4-D3 -B -q 0.01 --outdir TC1-A-H3K4-D3

</code></pre>

<p>In case <code>macs2</code> gives <code>command not found</code>, your are certainly missing the module, please see the <a href="http://ginolhac.github.io/chip-seq/install/">set-up</a></p>
<h3 id="check-model-inferred-by-macs2">check model inferred by MACS2</h3>
<p>execute R script.</p>
<pre><code>Rscript TC1-A-H3K4-D3/TC1-A-H3K4-D3_model.r
Rscript TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_model.r
</code></pre>

<p>fetch the pdf produced.</p>
<h3 id="sort-per-chromosomes-and-coordinates">sort per chromosomes and coordinates</h3>
<pre><code>find TC* -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;

</code></pre>

<h3 id="convert-to-bigwig">convert to bigwig</h3>
<p>in order to get smaller files</p>
<pre><code>find TC* -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>

<h3 id="fetch-the-files-and-display-them-in-igv">Fetch the files and display them in IGV</h3>
<p>IGV can be <a href="http://software.broadinstitute.org/software/igv/">downloaded</a> from the broadinstitute. </p>
<h3 id="perform-peak-calling-with-broad-option">Perform peak calling with broad option</h3>
<pre><code>macs2 callpeak -t TC1-H3K27-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-ST2-H3K27-D0-broad -B -q 0.01 --outdir TC1-ST2-H3K27-D0-broad &amp;
macs2 callpeak -t TC1-H3K27-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-A-H3K27-D3-broad -B -q 0.01 --outdir TC1-A-H3K27-D3-broad

</code></pre>

<p>Get the <code>bigwig</code> files for <strong>H3K27</strong>.
Redo those sort and conversion steps but only for the folders that end with 'broad'</p>
<pre><code>find TC*broad -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;
find TC*broad -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /work/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>

<h2 id="great-analysis">GREAT analysis</h2>
<p>The website <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a> allows pasting bed regions of enriched regions.</p>
<h3 id="predict-functions-of-cis-regulatory-regions">predict functions of cis-regulatory regions</h3>
<p>Using the <code>TC1-A-H3K4_peaks.narrowPeak</code> file produced by MACS2.</p>
<p>This file has the different fields:</p>
<ol>
<li>chromosome</li>
<li>start</li>
<li>end</li>
<li>peak name</li>
<li>integer score for display</li>
<li>strand</li>
<li>fold-change</li>
<li>-log<sub>10</sub> pvalue</li>
<li>-log<sub>10</sub> qvalue</li>
<li>relative summit position to peak start</li>
</ol>
<p>Let's format the file as a 3 fields BED file and focus on more significant peaks filtering on <em>q-values</em>.</p>
<pre><code>awk '$9&gt;40'  TC1-A-H3K4/TC1-A-H3K4_peaks.narrowPeak | cut -f 1-3 | sed 's/^/chr/' &gt;  TC1-A-H3K4/TC1-A-H3K4_peaks.bed
cat TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broadPeak | cut -f 1-3 | sed 's/^/chr/' &gt; TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broad.bed

</code></pre>

<p>then  </p>
<ul>
<li>load the BED in <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a>  </li>
<li>for the relevant genome, <code>mm10</code>  </li>
<li>association rule:<ul>
<li><em>Single nearest gene</em> for <strong>H3K4</strong> </li>
<li><em>Two nearest genes</em> for <strong>H3K27</strong> </li>
</ul>
</li>
</ul>
<h3 id="differential-peak-calling">Differential peak calling</h3>
<p><a href="http://www.regulatory-genomics.org/odin-2/basic-introduction/">ODIN</a> allows comparing two conditions associated with their own controls.</p>
<p>A command line looks like</p>
<pre><code>rgt-ODIN  --input-1=../TC1-I-ST2-D0.GRCm38.p3.q30.bam \
          --input-2=../TC1-I-A-D3.GRCm38.p3.q30.bam \
          -m -n TC1-I-A-D0vsD15 -v \
          TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam TC1-H3K4-A-D3.GRCm38.p3.q30.bam \
          ../references/GRCm38.p3.fasta ../references/GRCm38.p3.chom.sizes

</code></pre>
              
            </div>
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/ginolhac/chip-seq/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
