
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="ChIP-seq tutorial, University of Luxembourg, Department Life Sciences and Medecine">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.2, mkdocs-material-7.2.4">
    
    
      
        <title>Chip seq - ChIP-seq tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.f7f47774.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#workflow" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq tutorials" class="md-header__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq tutorials
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Chip seq
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq tutorials" class="md-nav__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    ChIP-seq tutorials
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../cli/" class="md-nav__link">
        Command line, basics
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../snakemake/" class="md-nav__link">
        Setup snakemake
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../workflow.md" class="md-nav__link">
        Workflow ChIP-seq
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../mapping/" class="md-nav__link">
        Mapping
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../peak/" class="md-nav__link">
        Peak calling
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../contact/" class="md-nav__link">
        Contact
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#workflow" class="md-nav__link">
    Workflow
  </a>
  
    <nav class="md-nav" aria-label="Workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#singularity-containers" class="md-nav__link">
    Singularity containers
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#book-resources-on-iris" class="md-nav__link">
    Book resources on iris
  </a>
  
    <nav class="md-nav" aria-label="Book resources on iris">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#optional" class="md-nav__link">
    (Optional)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#install-the-snakemake-template" class="md-nav__link">
    Install the snakemake template
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fetch-test-datasets" class="md-nav__link">
    Fetch test datasets
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-necessary-tools" class="md-nav__link">
    Load necessary tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#test-the-workflow" class="md-nav__link">
    Test the workflow
  </a>
  
    <nav class="md-nav" aria-label="Test the workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#config-files" class="md-nav__link">
    Config files
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-the-test-data-workflow" class="md-nav__link">
    Run the test-data workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#submit-passive-jobs" class="md-nav__link">
    Submit passive jobs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/ginolhac/chip-seq/edit/master/docs/chip-seq.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  <h1>Chip seq</h1>
                
                <h2 id="workflow">Workflow</h2>
<p>The workflow of all steps is summarised below:</p>
<p><img alt="" src="https://rawgit.com/ginolhac/chip-seq/master/workflow.png" /></p>
<p>and the template is hosted on the <a href="https://gitlab.lcsb.uni.lu/aurelien.ginolhac/snakemake-chip-seq">LCSB Gitlab</a></p>
<h3 id="singularity-containers">Singularity containers</h3>
<p><a href="https://www.sylabs.io/">Singularity</a> allows to use containers (from <em>i.e</em> <a href="https://www.docker.com/">Docker</a>) on High-Performance Computer.
For more details see the lecture <a href="https://ulhpc-tutorials.readthedocs.io/en/latest/containers/singularity/">by HPC team</a></p>
<p>Shortly, a container with all the necessary tools, softwares and all libraries are embeded. Hence, you need to book the HPC resources but not install those softwares. <strong>Snakemake</strong> will load the container for every  <strong>action</strong> it runs.</p>
<h3 id="book-resources-on-iris">Book resources on iris</h3>
<p><code>si</code> is a shortcut for booking a short <strong>interactive</strong> session (30 minutes, 1 core).</p>
<p>But, we can alter the time for one hour with <code>-t</code> and ask for more cores with <code>-c</code>. Example below where we ask for 1 jour and 6 cores:</p>
<pre><code>si -t 1:00:00 -c 6
</code></pre>
<p>Interactive sessions are limited to maximum 2 hours. See at the bottom how to submit <strong>passive</strong> jobs (max walltime: 48 hours).</p>
<h4 id="optional">(Optional)</h4>
<p>Two nodes were reserved for this course. This will work if you have a student accounts or if you are part of the Department of Life Sciences.
To access them you need to add the following parameter for Thursday:</p>
<pre><code>si -t 1:0:0 -c 6 --reservation=aginolhac-teaching-20210902
</code></pre>
<p>and for Friday:</p>
<pre><code>si -t 1:0:0 -c 6 --reservation=aginolhac-teaching-20210903
</code></pre>
<h3 id="install-the-snakemake-template">Install the snakemake template</h3>
<p>We will work in the <code>scratch</code> partition.</p>
<ul>
<li>the alias to go there is: </li>
</ul>
<pre><code class="language-bash">cds
</code></pre>
<p>Type <code>pwd</code> to ensure you are in <code>/scratch/users/username</code></p>
<ul>
<li>create the directory and go inside:</li>
</ul>
<pre><code class="language-bash">mkdir snakemake-chip-seq
cd snakemake-chip-seq
</code></pre>
<p>and run the following commands:</p>
<pre><code class="language-bash">VERSION=&quot;v0.1.1&quot;
wget -qO- https://gitlab.lcsb.uni.lu/aurelien.ginolhac/snakemake-chip-seq/-/archive/${VERSION}/snakemake-chip-seq-${VERSION}.tar.gz | tar xfz - --strip-components=1
</code></pre>
<p>this command will download, extract (without the root folder) the following files:</p>
<pre><code>CHANGELOG.md
config/
Dockerfile
LICENSE
README.md
resources/
workflow/
</code></pre>
<p>You can check the list of files using <code>ll</code> (alias of long list: <code>ls -l</code>)</p>
<p>you may want to delete the <code>LICENSE</code>, <code>Dockerfile</code>, <code>CHANGELOG.md</code> and <code>README.md</code> if you wish, 
they are not used by <code>snakemake</code> for runtime.</p>
<h3 id="fetch-test-datasets">Fetch test datasets</h3>
<p>Using the <a href="https://github.com/nf-core/test-datasets">nextflow datasets</a>, clone it using <code>git</code>:</p>
<pre><code class="language-bash">git clone -b chipseq --depth 1 https://github.com/nf-core/test-datasets.git
</code></pre>
<h3 id="load-necessary-tools">Load necessary tools</h3>
<ul>
<li>load <code>singularity</code></li>
</ul>
<p>the command once on a <code>node</code> is:</p>
<pre><code class="language-bash">(base) user@access module load tools/Singularity
</code></pre>
<ul>
<li>load the <code>snakemake</code> environment</li>
</ul>
<pre><code class="language-bash">(base) user@access $ conda activate snakemake
</code></pre>
<p>Of note, the 2 above steps can be replaced by the alias <code>smk</code> if you added the alias in your <code>.bashrc</code></p>
<p>It should look like this from accessing the access machine to getting the resources and activating the environment:</p>
<pre><code class="language-bash">(base) aginolhac@access1.iris-cluster.uni.lux(14:05:02)-&gt; 20:56): ~ $ si -c 6 -t 1:00:00
# salloc -p interactive --qos debug -C batch 
salloc: Pending job allocation 2424900
salloc: job 2424900 queued and waiting for resources
salloc: job 2424900 has been allocated resources
salloc: Granted job allocation 2424900
salloc: Waiting for resource configuration
salloc: Nodes iris-139 are ready for job
(base) aginolhac@iris-139(14:17:21)-&gt; 29:51)(2424900 1N/T/1CN): ~ $ smk
(snakemake) aginolhac@iris-139(14:17:23)-&gt; 29:49)(2424900 1N/T/1CN): ~ $
</code></pre>
<h3 id="test-the-workflow">Test the workflow</h3>
<h4 id="config-files">Config files</h4>
<p>3 tabulated separated values file (<code>.tsv</code>) defined where are the data and how to process them. They are all in the <code>config</code> folder.
Everything in the <code>workflow</code> contains the snakemake machinery and can be ignored if you are not interested into this.</p>
<ul>
<li>units</li>
</ul>
<p>This file has the key column in the first column, that must match the first colummn of <code>samples.tsv</code>.
Then <code>unit</code> are for <strong>technical</strong> replicates. </p>
<p><code>fq1</code> is the path the first pair of <code>FASTQ</code> or for single-end. In this last case, <code>fq2</code> needs to be empty.</p>
<p>The 2 last columns, <code>sra_accession</code> and <code>platform</code> are not going to be used.</p>
<p>Display the content of <code>config/units.tsv</code> (using <code>less config/units.tsv</code> for example) it should look like:</p>
<pre><code>sample  unit    fq1 fq2 sra_accession   platform
Spt5_IN 1   test-datasets/testdata/SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k_R1.fastq.gz   test-datasets/testdata/SRR5204809_Spt5-ChIP_Input1_SacCer_ChIP-Seq_ss100k_R2.fastq.gz       ILLUMINA
Spt5_IN 2   test-datasets/testdata/SRR5204810_Spt5-ChIP_Input2_SacCer_ChIP-Seq_ss100k_R1.fastq.gz   test-datasets/testdata/SRR5204810_Spt5-ChIP_Input2_SacCer_ChIP-Seq_ss100k_R2.fastq.gz       ILLUMINA
Spt5    1   test-datasets/testdata/SRR5204807_Spt5-ChIP_IP1_SacCer_ChIP-Seq_ss100k_R1.fastq.gz  test-datasets/testdata/SRR5204807_Spt5-ChIP_IP1_SacCer_ChIP-Seq_ss100k_R2.fastq.gz      ILLUMINA
Spt5    2   test-datasets/testdata/SRR5204808_Spt5-ChIP_IP2_SacCer_ChIP-Seq_ss100k_R1.fastq.gz  test-datasets/testdata/SRR5204808_Spt5-ChIP_IP2_SacCer_ChIP-Seq_ss100k_R2.fastq.gz      ILLUMINA
</code></pre>
<ul>
<li>samples</li>
</ul>
<p>Now look at <code>config/samples.tsv</code>:</p>
<pre><code>sample  group   batch_effect    control antibody    peak-analysis
Spt5_IN SptA    batch1      Spt 
Spt5    SptA    batch1  Spt5_IN Spt narrow
</code></pre>
<p><code>sample</code> are ID that must match the <code>units.tsv</code> <code>sample</code> column. </p>
<p><code>group</code> are for biological replicate, <code>batch_effect</code> is self-explanatory.</p>
<p><code>control</code> allows to specify which <strong>input</strong> has to be used. Leave it empty for input DNA sample.</p>
<p><code>antibody</code> is for antobody or histone mark used.</p>
<p><code>peak-analysis</code> either <code>narrow</code> (for K4) or <code>broad</code> (K27 or K36). This will be used when running <code>macs2</code>.</p>
<ul>
<li>config</li>
</ul>
<p>The file <code>config.yaml</code> contains the generic parameters for the analysis.</p>
<p>More specifically, it allows to specify the <strong>reference genome</strong> to use.</p>
<p>Here the section is designed for the <strong>yeast</strong> genome, that fits the test-data</p>
<pre><code>resources:
  ref:
    # Ensembl species name
    species: saccharomyces_cerevisiae
    # Ensembl release
    release: 101
    # Genome build
    build: R64-1-1
    # for testing data a single chromosome can be selected (leave empty for a regular analysis)
    chromosome:
    # specify release version number of igenomes list to use (see https://github.com/nf-core/chipseq/releases), e.g. 1.2.2
    igenomes_release: 1.2.2
    # if igenomes.yaml cannot be used, a value for the mappable or effective genome size can be specified here, e.g. macs-gsize: 2.7e9
    macs-gsize:
    # if igenomes.yaml cannot be used, a path to an own blacklist can be specified here
    blacklist:
</code></pre>
<p>This section will need to be updated when we use the real human data</p>
<h4 id="run-the-test-data-workflow">Run the test-data workflow</h4>
<ol>
<li>Check that your prompt is indicating that you are on a computing node, and with the snakemake environment loaded.</li>
</ol>
<p>Something like that:</p>
<pre><code>(snakemake) 0 [student20@iris-114 student20](2469485 1N/T/1CN)$
</code></pre>
<ol>
<li>Dry-run</li>
</ol>
<pre><code>snakemake -n
</code></pre>
<pre><code>[Wed Sep  1 10:53:50 2021]
localrule all:
    input: &lt;TBD&gt;
    jobid: 0
    resources: tmpdir=/tmp

Job stats:
job                  count    min threads    max threads
-----------------  -------  -------------  -------------
all                      1              1              1
generate_igenomes        1              1              1
get_gsize                1              1              1
total                    3              1              1

This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
</code></pre>
<ol>
<li>Produce the Direct Acyclic Graph</li>
</ol>
<p>With the alias, it is <code>dag</code>. You can then fetch and look at the <code>dag.pdf</code>. All jobs are with a plain line, meaning they are not done.</p>
<ol>
<li>Run the real workflow</li>
</ol>
<pre><code class="language-bash">snakemake --use-singularity --singularity-args &quot;-B /scratch/users/aginolhac:/scratch/users/aginolhac&quot;  -j 6
</code></pre>
<p>First it download the <code>singularity</code> image (and will cache it). It takes a few minutes (feels long however).</p>
<p>Then you can see the workflow moving on. <strong>107</strong> jobs are scheduled.</p>
<p>Messages like</p>
<pre><code>System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to create bus connection: Host is down
</code></pre>
<p>can be ignored. The whole workflow takes ~ 15 minutes with 6 cores. </p>
<ol>
<li>Create the report</li>
</ol>
<pre><code>snakemake --report
</code></pre>
<p>You can now fetch both <code>report.html</code> and <code>results/qc/multiqc/multiqc.html</code> that summarise the run.</p>
<ol>
<li>DAG again</li>
</ol>
<p>if you wish, the lines are now dashed.</p>
<h3 id="submit-passive-jobs">Submit passive jobs</h3>
<p>It consists in two steps:</p>
<ul>
<li>create a file <code>launcher.sh</code> at the root of your snakemake that contains the following lines:</li>
</ul>
<pre><code class="language-bash">#!/bin/bash -l
#SBATCH -N 1
#SBATCH -J JOB_NAME
#SBATCH --mail-type=begin,end,fail
#SBATCH --mail-user=yours@mail.lu
#SBATCH --ntasks-per-node=1
#SBATCH --mem-per-cpu=4096
#SBATCH -c 12
#SBATCH --time=0-06:00:00
#SBATCH -p batch

module load tools/Singularity
conda activate snakemake

srun snakemake --use-singularity --singularity-args &quot;-B /scratch/users/aginolhac:/scratch/users/aginolhac&quot; -j 12

</code></pre>
<p>You need to update the <code>JOB_NAME</code> and your email address. You can also delete those 3 lines (line 3 to 5) if you are interested into this monitoring.</p>
<p>The duration (here 6 hours) and number of cores can be adapted to your needs.</p>
<ul>
<li>run the submission with </li>
</ul>
<pre><code class="language-bash">sbatch launcher.sh
</code></pre>
<p>your job will be queued and start whenever it's possible and according to the <a href="https://hpc-docs.uni.lu/slurm/fairsharing/">fairshare</a>.</p>
<p>You can monitor your scheduling with <code>squeue -la -u $USER</code></p>
<p>See also the HPC team <a href="https://ulhpc-tutorials.readthedocs.io/en/latest/basic/scheduling/">tutorial</a>.</p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.709b4209.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.febc23d1.min.js"></script>
      
    
  </body>
</html>