
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="ChIP-seq tutorial, University of Luxembourg, Department Life Sciences and Medecine">
      
      
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.2, mkdocs-material-7.2.4">
    
    
      
        <title>Peak calling - ChIP-seq tutorials</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.f7f47774.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      


    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#peak-calling" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="ChIP-seq tutorials" class="md-header__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            ChIP-seq tutorials
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Peak calling
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="ChIP-seq tutorials" class="md-nav__button md-logo" aria-label="ChIP-seq tutorials" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    ChIP-seq tutorials
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/ginolhac/chip-seq/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../cli/" class="md-nav__link">
        Command line, basics
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../snakemake/" class="md-nav__link">
        Setup snakemake
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../workflow/" class="md-nav__link">
        Workflow
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../mapping/" class="md-nav__link">
        Mapping
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Peak calling
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Peak calling
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#peak-calling" class="md-nav__link">
    Peak calling
  </a>
  
    <nav class="md-nav" aria-label="Peak calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-the-bam-files-with-all-chromosomes" class="md-nav__link">
    Retrieve the BAM files with all chromosomes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-peak-calling" class="md-nav__link">
    Perform peak calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-model-inferred-by-macs2" class="md-nav__link">
    check model inferred by MACS2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sort-per-chromosomes-and-coordinates" class="md-nav__link">
    sort per chromosomes and coordinates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#convert-to-bigwig" class="md-nav__link">
    convert to bigwig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fetch-the-files-and-display-them-in-igv" class="md-nav__link">
    Fetch the files and display them in IGV
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-peak-calling-with-broad-option" class="md-nav__link">
    Perform peak calling with broad option
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#great-analysis" class="md-nav__link">
    GREAT analysis
  </a>
  
    <nav class="md-nav" aria-label="GREAT analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#predict-functions-of-cis-regulatory-regions" class="md-nav__link">
    predict functions of cis-regulatory regions
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-with-ngsplot" class="md-nav__link">
    alternative with ngsplot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#differential-peak-calling" class="md-nav__link">
    Differential peak calling
  </a>
  
    <nav class="md-nav" aria-label="Differential peak calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    visualization
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meta-analysis-using-great" class="md-nav__link">
    meta-analysis using GREAT
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../contact/" class="md-nav__link">
        Contact
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#peak-calling" class="md-nav__link">
    Peak calling
  </a>
  
    <nav class="md-nav" aria-label="Peak calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#retrieve-the-bam-files-with-all-chromosomes" class="md-nav__link">
    Retrieve the BAM files with all chromosomes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-peak-calling" class="md-nav__link">
    Perform peak calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#check-model-inferred-by-macs2" class="md-nav__link">
    check model inferred by MACS2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sort-per-chromosomes-and-coordinates" class="md-nav__link">
    sort per chromosomes and coordinates
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#convert-to-bigwig" class="md-nav__link">
    convert to bigwig
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fetch-the-files-and-display-them-in-igv" class="md-nav__link">
    Fetch the files and display them in IGV
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-peak-calling-with-broad-option" class="md-nav__link">
    Perform peak calling with broad option
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#great-analysis" class="md-nav__link">
    GREAT analysis
  </a>
  
    <nav class="md-nav" aria-label="GREAT analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#predict-functions-of-cis-regulatory-regions" class="md-nav__link">
    predict functions of cis-regulatory regions
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#alternative-with-ngsplot" class="md-nav__link">
    alternative with ngsplot
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#differential-peak-calling" class="md-nav__link">
    Differential peak calling
  </a>
  
    <nav class="md-nav" aria-label="Differential peak calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#visualization" class="md-nav__link">
    visualization
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#meta-analysis-using-great" class="md-nav__link">
    meta-analysis using GREAT
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/ginolhac/chip-seq/edit/master/docs/peak.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                  <h1>Peak calling</h1>
                
                <h2 id="peak-calling">Peak calling</h2>
<p>Using <a href="https://github.com/taoliu/MACS/">MACS2</a></p>
<p>For both the day 0 and day 3 of differentiation into adipocytes, two files are available</p>
<ul>
<li>input, as control</li>
<li>histone modification H3K4</li>
</ul>
<p><strong>MACS2</strong> is going to use both files to normalize the read counts and perform the peak calling.</p>
<h3 id="retrieve-the-bam-files-with-all-chromosomes">Retrieve the BAM files with all chromosomes</h3>
<pre><code>cd ~/chip-seq
mkdir bams
cd bams
ln -s /scratch/users/aginolhac/chip-seq/bams/*.bam .
</code></pre>
<h3 id="perform-peak-calling">Perform peak calling</h3>
<pre><code>macs2 callpeak -t TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-ST2-H3K4-D0 -B -q 0.01 --outdir TC1-ST2-H3K4-D0 &amp;
macs2 callpeak -t TC1-H3K4-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-A-H3K4-D3 -B -q 0.01 --outdir TC1-A-H3K4-D3

</code></pre>
<p>In case <code>macs2</code> gives <code>command not found</code>, your are certainly missing the module, please see the <a href="http://ginolhac.github.io/chip-seq/install/">set-up</a></p>
<h3 id="check-model-inferred-by-macs2">check model inferred by MACS2</h3>
<p>execute R script.</p>
<pre><code>Rscript TC1-A-H3K4-D3/TC1-A-H3K4-D3_model.r
Rscript TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_model.r
</code></pre>
<p>fetch the pdf produced.</p>
<h3 id="sort-per-chromosomes-and-coordinates">sort per chromosomes and coordinates</h3>
<pre><code>find TC* -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;

</code></pre>
<h3 id="convert-to-bigwig">convert to bigwig</h3>
<p>in order to get smaller files</p>
<pre><code>find TC* -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>
<h3 id="fetch-the-files-and-display-them-in-igv">Fetch the files and display them in IGV</h3>
<p>IGV can be <a href="http://software.broadinstitute.org/software/igv/">downloaded</a> from the broadinstitute. </p>
<h3 id="perform-peak-calling-with-broad-option">Perform peak calling with broad option</h3>
<pre><code>macs2 callpeak -t TC1-H3K27-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-ST2-H3K27-D0-broad -B -q 0.01 --outdir TC1-ST2-H3K27-D0-broad &amp;
macs2 callpeak -t TC1-H3K27-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-A-H3K27-D3-broad -B -q 0.01 --outdir TC1-A-H3K27-D3-broad

</code></pre>
<p>Get the <code>bigwig</code> files for <strong>H3K27</strong>.
Redo those sort and conversion steps but only for the folders that end with 'broad'</p>
<pre><code>find TC*broad -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;
find TC*broad -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>
<h2 id="great-analysis">GREAT analysis</h2>
<p>The website <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a> allows pasting bed regions of enriched regions.</p>
<h3 id="predict-functions-of-cis-regulatory-regions">predict functions of cis-regulatory regions</h3>
<p>Using the <code>TC1-A-H3K4-D3_peaks.narrowPeak</code> file produced by MACS2.</p>
<p>This file has the different fields:</p>
<ol>
<li>chromosome</li>
<li>start</li>
<li>end</li>
<li>peak name</li>
<li>integer score for display</li>
<li>strand</li>
<li>fold-change</li>
<li>-log<sub>10</sub> pvalue</li>
<li>-log<sub>10</sub> qvalue</li>
<li>relative summit position to peak start</li>
</ol>
<p>Let's format the file as a 3 fields BED file and focus on more significant peaks filtering on <em>q-values</em>.</p>
<pre><code>awk '$9&gt;40'  TC1-A-H3K4-D3/TC1-A-H3K4-D3_peaks.narrowPeak | cut -f 1-3 | sed 's/^/chr/' &gt;  TC1-A-H3K4-D3/TC1-A-H3K4_peaks.bed
awk '$9&gt;40'  TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_peaks.narrowPeak | cut -f 1-3 | sed 's/^/chr/' &gt;  TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_peaks.bed
</code></pre>
<p>For H3K27:</p>
<pre><code>cat TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broadPeak | cut -f 1-3 | sed 's/^/chr/' &gt; TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broad.bed

</code></pre>
<p>then  </p>
<ul>
<li>load the BED in <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a>  </li>
<li>for the relevant genome, <code>mm10</code>  </li>
<li>association rule:<ul>
<li><em>Single nearest gene</em> for <strong>H3K4</strong> </li>
<li><em>Two nearest genes</em> for <strong>H3K27</strong> </li>
</ul>
</li>
</ul>
<h3 id="alternative-with-ngsplot">alternative with ngsplot</h3>
<p>Example of <a href="https://github.com/shenlab-sinai/ngsplot">ngsplot</a> where gene expression ranked the genes from top to bottom and ChIP-seq of H3K4 is mapped with the red density on top.</p>
<p><img alt="" src="https://raw.githubusercontent.com/shenlab-sinai/ngsplot/develop/webimgs/hesc.H3k4me3.tss.all.png" /></p>
<h3 id="differential-peak-calling">Differential peak calling</h3>
<p><a href="http://www.regulatory-genomics.org/thor-2/basic-intrstruction/">THOR</a> allows comparing two conditions associated with their own controls and with replicates.</p>
<ul>
<li>first, index the bams</li>
</ul>
<pre><code>parallel &quot;samtools index {}&quot; ::: *bam
</code></pre>
<ul>
<li>second, create a config file <code>THOR.config</code> that contains:</li>
</ul>
<pre><code>#rep1
TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam
#rep2
TC1-H3K4-A-D3.GRCm38.p3.q30.bam
#chrom_sizes
/scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes
#genome
/scratch/users/aginolhac/chip-seq/references/GRCm38.p3.fasta
#inputs1
TC1-I-ST2-D0.GRCm38.p3.q30.bam
#inputs2
TC1-I-A-D3.GRCm38.p3.q30.bam
</code></pre>
<p>A command line looks like</p>
<pre><code>rgt-THOR -m -n TC1-I-A-D0vsD3 --output-dir=TC1-I-A-D0vsD3 THOR.config
</code></pre>
<p>takes ~ 25 minutes</p>
<h4 id="visualization">visualization</h4>
<ul>
<li>load the file <code>TC1-I-A-D0vsD3-diffpeaks.bed</code> and the bigwig files (<code>.bw</code> extension)</li>
<li>color bigwig for D0 in red</li>
<li>color bigwig for D3 in green</li>
<li>select both bigwig and right-click to <strong>Overlay tracks</strong></li>
<li>the BED track should display in red the regions with higher enrichments in the D0, green in the D3.</li>
</ul>
<h4 id="meta-analysis-using-great">meta-analysis using GREAT</h4>
<p>You can play with the BED file in R with this code to extract the fold-change from counts.
It is encoded in the 11th field of the narrowPeak file as <strong>counts</strong> for the first condition (D0-ST2) and counts for the second condition (D3-A)</p>
<pre><code># load the file using the tidyverse
library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
diffpeaks &lt;- read_tsv(&quot;TC1-I-A-D0vsD3-diffpeaks.bed&quot;,
                      col_names = FALSE, trim_ws = TRUE, col_types = cols(X1 = col_character()))
# split the last field into three
diffpeaks %&gt;%
  separate(X11, into = c(&quot;count1&quot;, &quot;count2&quot;, &quot;third&quot;), sep = &quot;;&quot;, convert = TRUE) %&gt;%
  mutate(FC = count2 / count1) -&gt; thor_splitted
# plot the histogram of the fold-change computed above, count second condition / count 1st condition
thor_splitted %&gt;%
  ggplot(aes(x = log2(FC))) +
  geom_histogram() +
  scale_x_continuous(breaks = seq(-5, 3, 1))

# create a bed file, append chr to chromosome names and write down the file
thor_splitted %&gt;%
  filter(log2(FC) &gt; 0.5) %&gt;%
  select(X1, X2, X3) %&gt;%
  mutate(X1 = paste0(&quot;chr&quot;, X1)) %&gt;%
  write_tsv(&quot;THOR_logFC0.5.bed&quot;, col_names = FALSE)
</code></pre>
<p>you can now import the file <code>THOR_logFC0.5.bed</code> into GREAT and see again how the meta-analysis looks like.</p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../mapping/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Mapping" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Mapping
            </div>
          </div>
        </a>
      
      
        
        <a href="../contact/" class="md-footer__link md-footer__link--next" aria-label="Next: Contact" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Contact
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../assets/javascripts/workers/search.709b4209.min.js", "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.febc23d1.min.js"></script>
      
    
  </body>
</html>