<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Peak calling - ChIP-seq tutorials</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Peak calling";
    var mkdocs_page_input_path = "peak.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> ChIP-seq tutorials</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../install/">Setup</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../cli/">Command line, basics</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../fastqc/">QC</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../mapping/">Mapping</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Peak calling</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#retrieve-the-bam-files-with-all-chromosomes">Retrieve the BAM files with all chromosomes</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#perform-peak-calling">Perform peak calling</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#check-model-inferred-by-macs2">check model inferred by MACS2</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#sort-per-chromosomes-and-coordinates">sort per chromosomes and coordinates</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#convert-to-bigwig">convert to bigwig</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#fetch-the-files-and-display-them-in-igv">Fetch the files and display them in IGV</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#perform-peak-calling-with-broad-option">Perform peak calling with broad option</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../public/">Public data</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../contact/">Contact</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">ChIP-seq tutorials</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Peak calling</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ginolhac/chip-seq/edit/master/docs/peak.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h2 id="peak-calling">Peak calling</h2>
<p>Using <a href="https://github.com/taoliu/MACS/">MACS2</a></p>
<p>For both the day 0 and day 3 of differentiation into adipocytes, two files are available</p>
<ul>
<li>input, as control</li>
<li>histone modification H3K4</li>
</ul>
<p><strong>MACS2</strong> is going to use both files to normalize the read counts and perform the peak calling.</p>
<h3 id="retrieve-the-bam-files-with-all-chromosomes">Retrieve the BAM files with all chromosomes</h3>
<pre><code>cd ~/chip-seq
mkdir bams
cd bams
ln -s /scratch/users/aginolhac/chip-seq/bams/*.bam .
</code></pre>
<h3 id="perform-peak-calling">Perform peak calling</h3>
<pre><code>macs2 callpeak -t TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-ST2-H3K4-D0 -B -q 0.01 --outdir TC1-ST2-H3K4-D0 &amp;
macs2 callpeak -t TC1-H3K4-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM -g mm -n TC1-A-H3K4-D3 -B -q 0.01 --outdir TC1-A-H3K4-D3

</code></pre>
<p>In case <code>macs2</code> gives <code>command not found</code>, your are certainly missing the module, please see the <a href="http://ginolhac.github.io/chip-seq/install/">set-up</a></p>
<h3 id="check-model-inferred-by-macs2">check model inferred by MACS2</h3>
<p>execute R script.</p>
<pre><code>Rscript TC1-A-H3K4-D3/TC1-A-H3K4-D3_model.r
Rscript TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_model.r
</code></pre>
<p>fetch the pdf produced.</p>
<h3 id="sort-per-chromosomes-and-coordinates">sort per chromosomes and coordinates</h3>
<pre><code>find TC* -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;

</code></pre>
<h3 id="convert-to-bigwig">convert to bigwig</h3>
<p>in order to get smaller files</p>
<pre><code>find TC* -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>
<h3 id="fetch-the-files-and-display-them-in-igv">Fetch the files and display them in IGV</h3>
<p>IGV can be <a href="http://software.broadinstitute.org/software/igv/">downloaded</a> from the broadinstitute. </p>
<h3 id="perform-peak-calling-with-broad-option">Perform peak calling with broad option</h3>
<pre><code>macs2 callpeak -t TC1-H3K27-ST2-D0.GRCm38.p3.q30.bam \
               -c TC1-I-ST2-D0.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-ST2-H3K27-D0-broad -B -q 0.01 --outdir TC1-ST2-H3K27-D0-broad &amp;
macs2 callpeak -t TC1-H3K27-A-D3.GRCm38.p3.q30.bam \
               -c TC1-I-A-D3.GRCm38.p3.q30.bam \
               -f BAM --broad -g mm -n TC1-A-H3K27-D3-broad -B -q 0.01 --outdir TC1-A-H3K27-D3-broad

</code></pre>
<p>Get the <code>bigwig</code> files for <strong>H3K27</strong>.
Redo those sort and conversion steps but only for the folders that end with 'broad'</p>
<pre><code>find TC*broad -name '*.bdg' | parallel &quot;sort -k1,1 -k2,2n {} &gt; {.}.sort.bdg&quot;
find TC*broad -name '*sort.bdg' | parallel -j 2 &quot;bedGraphToBigWig {} \
  /scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes {.}.bigwig&quot;

</code></pre>
<h2 id="great-analysis">GREAT analysis</h2>
<p>The website <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a> allows pasting bed regions of enriched regions.</p>
<h3 id="predict-functions-of-cis-regulatory-regions">predict functions of cis-regulatory regions</h3>
<p>Using the <code>TC1-A-H3K4-D3_peaks.narrowPeak</code> file produced by MACS2.</p>
<p>This file has the different fields:</p>
<ol>
<li>chromosome</li>
<li>start</li>
<li>end</li>
<li>peak name</li>
<li>integer score for display</li>
<li>strand</li>
<li>fold-change</li>
<li>-log<sub>10</sub> pvalue</li>
<li>-log<sub>10</sub> qvalue</li>
<li>relative summit position to peak start</li>
</ol>
<p>Let's format the file as a 3 fields BED file and focus on more significant peaks filtering on <em>q-values</em>.</p>
<pre><code>awk '$9&gt;40'  TC1-A-H3K4-D3/TC1-A-H3K4-D3_peaks.narrowPeak | cut -f 1-3 | sed 's/^/chr/' &gt;  TC1-A-H3K4-D3/TC1-A-H3K4_peaks.bed
awk '$9&gt;40'  TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_peaks.narrowPeak | cut -f 1-3 | sed 's/^/chr/' &gt;  TC1-ST2-H3K4-D0/TC1-ST2-H3K4-D0_peaks.bed
</code></pre>
<p>For H3K27:</p>
<pre><code>cat TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broadPeak | cut -f 1-3 | sed 's/^/chr/' &gt; TC1-A-H3K27-D3-broad/TC1-A-H3K27-D3-broad_peaks.broad.bed

</code></pre>
<p>then  </p>
<ul>
<li>load the BED in <a href="http://bejerano.stanford.edu/great/public/html/">GREAT</a>  </li>
<li>for the relevant genome, <code>mm10</code>  </li>
<li>association rule:<ul>
<li><em>Single nearest gene</em> for <strong>H3K4</strong> </li>
<li><em>Two nearest genes</em> for <strong>H3K27</strong> </li>
</ul>
</li>
</ul>
<h3 id="alternative-with-ngsplot">alternative with ngsplot</h3>
<p>Example of <a href="https://github.com/shenlab-sinai/ngsplot">ngsplot</a> where gene expression ranked the genes from top to bottom and ChIP-seq of H3K4 is mapped with the red density on top.</p>
<p><img alt="" src="https://raw.githubusercontent.com/shenlab-sinai/ngsplot/develop/webimgs/hesc.H3k4me3.tss.all.png" /></p>
<h3 id="differential-peak-calling">Differential peak calling</h3>
<p><a href="http://www.regulatory-genomics.org/thor-2/basic-intrstruction/">THOR</a> allows comparing two conditions associated with their own controls and with replicates.</p>
<ul>
<li>first, index the bams</li>
</ul>
<pre><code>parallel &quot;samtools index {}&quot; ::: *bam
</code></pre>
<ul>
<li>second, create a config file <code>THOR.config</code> that contains:</li>
</ul>
<pre><code>#rep1
TC1-H3K4-ST2-D0.GRCm38.p3.q30.bam
#rep2
TC1-H3K4-A-D3.GRCm38.p3.q30.bam
#chrom_sizes
/scratch/users/aginolhac/chip-seq/references/GRCm38.p3.chom.sizes
#genome
/scratch/users/aginolhac/chip-seq/references/GRCm38.p3.fasta
#inputs1
TC1-I-ST2-D0.GRCm38.p3.q30.bam
#inputs2
TC1-I-A-D3.GRCm38.p3.q30.bam
</code></pre>
<p>A command line looks like</p>
<pre><code>rgt-THOR -m -n TC1-I-A-D0vsD3 --output-dir=TC1-I-A-D0vsD3 THOR.config
</code></pre>
<p>takes ~ 25 minutes</p>
<h4 id="visualization">visualization</h4>
<ul>
<li>load the file <code>TC1-I-A-D0vsD3-diffpeaks.bed</code> and the bigwig files (<code>.bw</code> extension)</li>
<li>color bigwig for D0 in red</li>
<li>color bigwig for D3 in green</li>
<li>select both bigwig and right-click to <strong>Overlay tracks</strong></li>
<li>the BED track should display in red the regions with higher enrichments in the D0, green in the D3.</li>
</ul>
<h4 id="meta-analysis-using-great">meta-analysis using GREAT</h4>
<p>You can play with the BED file in R with this code to extract the fold-change from counts.
It is encoded in the 11th field of the narrowPeak file as <strong>counts</strong> for the first condition (D0-ST2) and counts for the second condition (D3-A)</p>
<pre><code># load the file using the tidyverse
library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
diffpeaks &lt;- read_tsv(&quot;TC1-I-A-D0vsD3-diffpeaks.bed&quot;,
                      col_names = FALSE, trim_ws = TRUE, col_types = cols(X1 = col_character()))
# split the last field into three
diffpeaks %&gt;%
  separate(X11, into = c(&quot;count1&quot;, &quot;count2&quot;, &quot;third&quot;), sep = &quot;;&quot;, convert = TRUE) %&gt;%
  mutate(FC = count2 / count1) -&gt; thor_splitted
# plot the histogram of the fold-change computed above, count second condition / count 1st condition
thor_splitted %&gt;%
  ggplot(aes(x = log2(FC))) +
  geom_histogram() +
  scale_x_continuous(breaks = seq(-5, 3, 1))

# create a bed file, append chr to chromosome names and write down the file
thor_splitted %&gt;%
  filter(log2(FC) &gt; 0.5) %&gt;%
  select(X1, X2, X3) %&gt;%
  mutate(X1 = paste0(&quot;chr&quot;, X1)) %&gt;%
  write_tsv(&quot;THOR_logFC0.5.bed&quot;, col_names = FALSE)
</code></pre>
<p>you can now import the file <code>THOR_logFC0.5.bed</code> into GREAT and see again how the meta-analysis looks like.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../public/" class="btn btn-neutral float-right" title="Public data">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../mapping/" class="btn btn-neutral" title="Mapping"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/ginolhac/chip-seq/" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../mapping/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../public/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
